src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -510,57 +511,72 @@ private void copyRecipesAndArtifacts(Deployment deployment) throws InvalidReques
         }
     }
 
-    @SuppressWarnings("PMD.ExceptionAsFlowControl")
     private void copyRecipesToComponentStore(Path from) throws IOException {
         try (Stream<Path> files = Files.walk(from)) {
             for (Path r : files.collect(Collectors.toList())) {
```

Comment:
Can you add a skip for directories? We currently get a false error that some directory is not a recipe, which we can safely ignore.